debugger/src/main/kotlin/org/rust/debugger/RsBackendConsoleInjectionHelper.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.debugger
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.ModalityState
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiLanguageInjectionHost
+import com.intellij.xdebugger.XDebugSession
+import com.intellij.xdebugger.XDebugSessionListener
+import com.jetbrains.cidr.execution.debugger.BackendConsoleInjectionHelper
+import com.jetbrains.cidr.execution.debugger.CidrDebugProcess
+import com.jetbrains.cidr.execution.debugger.backend.lang.GDBExpressionPlaceholder
+import org.rust.lang.RsDebugInjectionListener
+import org.rust.lang.core.psi.ext.ancestorOrSelf
+
+class RsBackendConsoleInjectionHelper : BackendConsoleInjectionHelper {
+    override fun subscribeToInjection(session: XDebugSession) {
+        val connection = session.project.messageBus.connect()
+
+        val listener = object : RsDebugInjectionListener, XDebugSessionListener {
+            @Volatile
+            private var document: Document? = null
+
+            override fun evalDebugContext(host: PsiLanguageInjectionHost, context: RsDebugInjectionListener.DebugContext) {
```

Comment:
This method doesn't use state of listener at all, so it can be moved into `RustParserDefinition`. In this case you can make `DebugContext.element` final.

Reply:
It cannot be moved because of using `CidrDebugProcess.DEBUG_PROCESS_KEY`